setuptools_rust/setuptools_ext.py

```diff
@@ -17,6 +20,45 @@
 
 
 def add_rust_extension(dist):
+    sdist_base_class = dist.cmdclass.get("sdist", sdist)
+    sdist_options = sdist_base_class.user_options.copy()
+    sdist_boolean_options = sdist_base_class.boolean_options.copy()
+    sdist_negative_opt = sdist_base_class.negative_opt.copy()
+    sdist_options.extend([
+        ('vendor-crates', None,
+         "vendor Rust crates"),
+        ('no-vendor-crates', None,
+         "don't vendor Rust crates."
+         "[default; enable with --vendor-crates]"),
+    ])
+    sdist_boolean_options.append('vendor-crates')
+    sdist_negative_opt['no-vendor-crates'] = 'vendor-crates'
+
+    class sdist_rust_extension(sdist_base_class):
+        user_options = sdist_options
+        boolean_options = sdist_boolean_options
+        negative_opt = sdist_negative_opt
+
+        def initialize_options(self):
+            super().initialize_options()
+            self.vendor_crates = 0
+
+        def get_file_list(self):
+            if self.vendor_crates:
+                base_dir = self.distribution.get_fullname()
+                vendor_path = os.path.join(base_dir, "vendored-crates")
```

Comment:
The default directory for ``cargo vendor`` is ``vendor``. Is there any particular reason to select a different directory name?

```
$ cargo vendor --help
...
ARGS:
    <path>    Where to vendor crates (`vendor` by default)
```

Reply:
I was thinking `vendor` is too easy to conflict with normal project files/directories.

Reply:
Changed to `.cargo/vendor`.